inc/plugins/namespace.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Sets DID-less plugin slug as active.
+ *
+ * @package FAIR
+ */
+
+namespace FAIR\Plugins;
+
+use function FAIR\Updater\get_packages;
+
+/**
+ * Bootstrap
+ *
+ * @return void
+ */
+function bootstrap() {
+	add_filter( 'option_active_plugins', __NAMESPACE__ . '\\set_as_active' );
+	add_filter( 'wp_admin_notice_markup', __NAMESPACE__ . '\\hide_notice', 10, 3 );
+
+	// just for testing.
+	if ( ! function_exists( 'is_plugin_active' ) ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+	}
+	if ( is_plugin_active( 'git-updater/git-updater.php' ) ) {
+		wp_admin_notice( 'Git Updater is active' );
+	}
+}
+
+/**
+ * Set FAIR plugins as active using DID-less slug.
+ *
+ * @param  array $active_plugins Array of active plugins.
+ *
+ * @return array
+ */
+function set_as_active( $active_plugins ) {
+	remove_filter( 'option_active_plugins', __NAMESPACE__ . '\\set_as_active' );
+	$packages = get_packages();
+	$plugins = $packages['plugins'] ?? [];
+	$plugins = array_map( 'plugin_basename', $plugins );
+	foreach ( $plugins as $plugin ) {
+		if ( is_plugin_active( $plugin ) ) {
+			$plugins[] = get_file_without_did_hash( $plugin );
```

Comment:
Instead of adding this to `$plugins`, then doing `array_merge()` after the loop, couldn't this be added directly to `$active_plugins`? The `array_unique()` is fine of course since it saves on in-loop `in_array()` checks / `array_flip()` flopping.